source/tests/system/nidcpower_driver_api_tests.cpp

```diff
@@ -140,6 +210,141 @@ TEST_F(NiDCPowerDriverApiTest, NiDCPowerReset_SendRequest_ResetCompletesSuccessf
+  request.mutable_vi()->set_id(GetSessionId());
+  request.set_channel_name(channel_list);
+  request.set_attribute_id(attribute_to_set);
+  request.set_attribute_value(expected_value);
+  dcpower::SetAttributeViInt32Response response;
+
+  ::grpc::Status status = GetStub()->SetAttributeViInt32(&context, request, &response);
+  EXPECT_TRUE(status.ok());
+  expect_api_success(response.status());
+
+  ViInt32 get_attribute_value = get_int32_attribute(channel_list, attribute_to_set);
+  EXPECT_EQ(expected_value, get_attribute_value);
+}
+
+TEST_F(NiDCPowerDriverApiTest, NiDCPowerSetViReal64Attribute_SendRequest_GetViReal64AttributeMatches)
+{
+  // Attribute 'NIDCPOWER_ATTRIBUTE_MEASURE_WHEN' is set to 'MEASURE_WHEN_NIDCPOWER_VAL_AUTOMATICALLY_AFTER_SOURCE_COMPLETE' 
+  // before setting attribute 'NIDCPOWER_ATTRIBUTE_SOURCE_DELAY'.
+  
+  const char* channel_list = "0";
+  const dcpower::NiDCPowerAttributes attribute_to_set_measurewhen = dcpower::NiDCPowerAttributes::NIDCPOWER_ATTRIBUTE_MEASURE_WHEN;
+  const ViInt32 expected_value_measurewhen = dcpower::MeasureWhen::MEASURE_WHEN_NIDCPOWER_VAL_AUTOMATICALLY_AFTER_SOURCE_COMPLETE;
+  ::grpc::ClientContext context_ViInt32;
+  dcpower::SetAttributeViInt32Request request_ViInt32;
+  request_ViInt32.mutable_vi()->set_id(GetSessionId());
+  request_ViInt32.set_channel_name(channel_list);
+  request_ViInt32.set_attribute_id(attribute_to_set_measurewhen);
+  request_ViInt32.set_attribute_value(expected_value_measurewhen);
+  dcpower::SetAttributeViInt32Response response_ViInt32;
+  ::grpc::Status status_ViInt32 = GetStub()->SetAttributeViInt32(&context_ViInt32, request_ViInt32, &response_ViInt32);
+  EXPECT_TRUE(status_ViInt32.ok());
+  expect_api_success(response_ViInt32.status());
+
+  const dcpower::NiDCPowerAttributes attribute_to_set_sourcedelay = dcpower::NiDCPowerAttributes::NIDCPOWER_ATTRIBUTE_SOURCE_DELAY;
+  const ViReal64 expected_value_sourcedelay = 2.516;
+  ::grpc::ClientContext context_ViReal64;
+  dcpower::SetAttributeViReal64Request request_ViReal64;
+  request_ViReal64.mutable_vi()->set_id(GetSessionId());
+  request_ViReal64.set_channel_name(channel_list);
+  request_ViReal64.set_attribute_id(attribute_to_set_sourcedelay);
+  request_ViReal64.set_attribute_value(expected_value_sourcedelay);
+  dcpower::SetAttributeViReal64Response response_ViReal64;
+  ::grpc::Status status_ViReal64 = GetStub()->SetAttributeViReal64(&context_ViReal64, request_ViReal64, &response_ViReal64);
+  EXPECT_TRUE(status_ViReal64.ok());
+  expect_api_success(response_ViReal64.status());
+  
+  ViReal64 get_attribute_value_sourcedelay = get_real64_attribute(channel_list, attribute_to_set_sourcedelay);
+  EXPECT_EQ(expected_value_sourcedelay, get_attribute_value_sourcedelay);
+}
+
+TEST_F(NiDCPowerDriverApiTest, NiDCPowerSetBoolAttribute_SendRequest_GetBoolAttributeMatches)
+{
+  // Attribute 'NIDCPOWER_ATTRIBUTE_MEASURE_WHEN' is set to 'MEASURE_WHEN_NIDCPOWER_VAL_AUTOMATICALLY_AFTER_SOURCE_COMPLETE' 
```

Comment:
Same question. Is there no other simple bool attribute?

Reply:
Same reason, this attribute is also going to be used in our example.

Reply:
I have added helper function to shorten the code.

Reply:
@ni-pkhare , the indentation rules for tests are like this:
If test name is in the format of Given_When_Then,
There should be a new line after Given code section.
And one new line after When code section.

So if test name is Given_When_Then, then test body should have exactly 2 new lines.

Can you reorganize the tests to follow that convention?